x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/notify_badge/notify_badge.tsx

```diff
@@ -91,6 +91,27 @@ export const RulesListNotifyBadge: React.FunctionComponent<RulesListNotifyBadgeP
     [isSnoozedUntil]
   );
 
+  const focusTrapButtonRef = useRef<HTMLButtonElement>(null);
+
+  // "onRuleChanged" fn triggers a full re-render of button component,
+  // preventing us from returning focus to the cell that initiated these changes.
+  // This method acts as a wrapper around "onRuleChanged" and ensures that focus is returned
+  // to the correct location after the main action is executed.
+  const onRuleChangedWrapped: RulesListNotifyBadgeProps['onRuleChanged'] = useMemo(
+    () =>
+      async (...args) => {
+        const returnFocusTimeout = 100;
+        try {
+          await onRuleChanged(...args);
+        } finally {
+          setTimeout(() => {
```

Comment:
What is the reason for adding this timeout?

Reply:
@cnasikas the rendering cycle after calling `await onRuleChanged(...args)` must complete for the reference to `focusTrapButtonRef` to be updated. It might work without `setTimeout`, but I can't be certain it will work stable

Reply:
Having a timeout in the callback without React being aware may cause various issues. Also, the timeout will not be cleaned if the user navigates to another page and the component unmounts. Moving the `focusTrapButtonRef.current?.focus()` to a `useEffect` that will trigger every time the rule is updated seems more appropriate. I am curious about the UX of a user using the mouse. Will it focus the button also for them?

Reply:
@cnasikas, thank you for your suggestion. At the moment, I'm not entirely certain if it will be effective, probably not. I'll need some time to consider alternative approaches. But for now not sure that we can fix that differently 

@1Copenut fyi

Reply:
@JiaweiWu proposed the usage of `requestAnimationFrame`. It is being used by the EUI library (https://github.com/elastic/eui/blob/main/packages/eui/src/components/inline_edit/inline_edit_form.tsx#L242) Also, could [`autofocus`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus) work?

Reply:
Thank you @cnasikas and @JiaweiWu. The option with requestAnimationFrame seems to be working.